src/stravalib/tests/unit/test_refresh_token.py

```diff
@@ -0,0 +1,123 @@
+"""A series of unit tests to ensure the refresh token operations
+work as expected whether the user has set things up or not.
+
+TODO:
+
+exchange_code_for_token
+* make sure that access token and token expires are set properly in the
+ - protocol.py:360-370 & 414-419 (how are these different?)
+
+ ## Client.py
+ * in client - make sure the access token, refresh value and expires at are all
+  set client.py:120-179
+* refresh access token: 258
+i think these will be integration tests to add to client?
+vs unit
+
+
+"""
+
+import logging
+import os
+import time
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+
+@pytest.mark.parametrize(
+    "token_expires, expected_result",
+    [
+        (time.time() - 10, True),  # Token expired
+        (time.time() + 10, False),
+    ],
+)
+def test_token_expired(apiv3_instance, token_expires, expected_result):
+    """Test the _token_expired method for both expired and valid tokens."""
+
+    apiv3_instance.token_expires = token_expires
+    assert apiv3_instance._token_expired() == expected_result
+
+
+# caplog is a builtin fixture in pytest that captures logging outputs
+def test_token_expires_is_none_logs_warning(apiv3_instance, caplog):
+    """Test that _token_expired logs a warning if token_expires is None."""
+    apiv3_instance.token_expires = None
+
+    with caplog.at_level(logging.WARNING):
+        assert not apiv3_instance._token_expired()
+        assert (
+            "Please make sure you've set client.token_expires" in caplog.text
+        )
+
+
+##### Test refresh_expired_token #####
+
+
+@pytest.mark.parametrize(
+    "token_expired, expected_call", [(True, 1), (False, 0)]
+)
+def test_refresh_expired_token_called(
+    apiv3_instance, token_expired, expected_call
+):
+    """Test that `refresh_access_token` method is called when token is expired
+    and not called when valid.
+    """
+
+    apiv3_instance.refresh_token = "dummy_refresh_token"
+
+    # Patch os get envt variables and populate envt vals
+    with patch.dict(
+        os.environ,
+        {"STRAVA_CLIENT_ID": "12345", "STRAVA_CLIENT_SECRET": "secret"},
+    ):
```

Comment:
```suggestion
```

this context manager is no longer needed because it only is needed in the _check_credentials helper (I think)